### ISSUE TITLE
Correction found 17 errors due build 

### ISSUE THREAD

**Issue by Ana-Green** [Virtual-Universe/Virtual-Universe-Early-Dev#23]
Correction Found 17 errors all related like this:

D:\Virtual-Universe-NXG-master\OpenSim\Region\Framework\Properties\AssemblyInfo.cs

// Version information for an assembly consists of the following four values:
//
//      Major Version
//      Minor Version 
//      Build Number
//      Revision
//
[assembly: AssemblyVersion("0.2.001.*")]
[assembly: AddinRoot(id: "OpenSim.Region.Framework", version: OpenSim.VersionInfo.Version)]

And this:

D:\Virtual-Universe-NXG-master\OpenSim\Server\Base\ServerUtils.cs

using OpenSim.Framework.Servers;

[assembly:AddinRoot("Robust", OpenSim.VersionInfo.Version)]
namespace OpenSim.Server.Base
{
    [TypeExtensionPoint(Path="/Robust/Connector", Name="RobustConnector")]
    public interface IRobustConnector
    {
        string ConfigName
        {
            get;
        }
These errors pointing all to this [assembly:AddinRoot("Robust", OpenSim.VersionInfo.Version)] or  this [assembly: AddinRoot(id: "OpenSim.Region.Framework", version: OpenSim.VersionInfo.Version)] 

And so it keeps going 17 times in total.

Reported by Ana Green

**Comment by emperorstarfinder** [comment]
Good evening,

By copying and pasting the file the error is pointing to doesn't help.  We need to see the actual error message to determine what the actual error is.  

I suspect this is more related to the project cache then there being an actual error as this is building correctly internally.  I recommend downloading a fresh copy of the repository and attempting to build.  This way we can rule out that there is an issue in the project cache.

Best Regards,
Emperor Starfinder
CEO
Founder
Governing Body Chair
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Grid Architecture Development, Maintenance
Second Galaxy Development Team

**Comment by Ana-Green** [comment]
Well that  are 17 files but all  do point  to the section [assembly: AddinRoot ..  
You want  me to copy all files in here? 

Regards,
Ana Green

**Comment by emperorstarfinder** [comment]
Just the error messages that visual studio is giving you is what we need.  we know what is in the files.  Always put the error message your getting as this will help us pinpoint what is going on quicker.

Best Regards,
Emperor Starfinder
CEO
Founder
Governing Body Chair
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Grid Architecture Development, Maintenance
Second Galaxy Development Team

**Comment by BruceStarfinder** [comment]
Good evening,

This really isn't an error, but for some strange reason some build tools such as Visual Studios and Mono Develop believe it is an error.  Its really just an annoyance caused by certain developers at OpenSimulator thinking that using the Mono.Addins for the assembly and versioninfo related things was a smart idea which it isn't in reality.  In the meantime I have reverted the commit until I can straighten this all out.

Best Regards,
Bruce Starfinder
Core Developer
Work Group member
Grid Architecture Development Work Group
Grid Architecture Development, Database Development, Maintenance
Second Galaxy Development Team

**Comment by emperorstarfinder** [comment]
@BruceStarfinder would the better answer be to look for the commit in OpenSim that added the mono addin requirement for VersionInfo and AssemblyInfo and revert that particular commit? or do you think that could potentially break something else?

Best Regards,
Emperor Starfinder
CEO
Founder
Governing Body Chair
Core Developer
Work Group Leader
Grid Architecture Development Work Group
Grid Architecture Development, Maintenance
Second Galaxy Development Team

**Comment by BruceStarfinder** [comment]
@emperorstarfinder That thought did cross my mind about 2 seconds after I reverted the commit and pushed it to the repository.  This was building when I pushed the commit initially to the repository but apparently it broke with recent commits.  I am going to look at that route further to see if we really need to be doing that and if we don't need to do that method then I will just get rid of that method and revert it to the way it was.  At which point we shouldn't have this problem with VersionInfo any longer.

Best Regards,
Bruce Starfinder
Core Developer
Work Group member
Grid Architecture Development Work Group
Grid Architecture Development, Database Development, Maintenance
Second Galaxy Development Team

**Comment by Ana-Green** [comment]
Lets  keep it simple, I did download the zip  file, unlock it, de-compress it and run prebuild.bat  that created the SLN OpenSim.sln file and the compile.bat (Batch) file, this Batch did  not run just a  black window and  opt out. So I used OpenSim.sln and started Microsoft Visual Studio 2017 and  simple  did hit build, that went  shortly  good then I  get  a  list of errors Like "Code	Description	Project	File	Line	Suppression State
CS0182	An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type	OpenSim.Region.Framework	D:\Virtual-Universe-NXG-master\OpenSim\Region\Framework\Properties\AssemblyInfo.cs	35	Active
And  so  in a  list:

 Code	Description	Project	File	Line	Suppression State
CS0182	An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type	OpenSim.Region.Framework	D:\Virtual-Universe-NXG-master\OpenSim\Region\Framework\Properties\AssemblyInfo.cs	35	Active

 Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.Framework.dll' could not be found	OpenSim.Region.ClientStack	D:\Virtual-Universe-NXG-master\OpenSim\Region\ClientStack\CSC	1	Active

 Code	Description	Project	File	Line	Suppression State
CS0182	An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type	OpenSim.Server.Base	D:\Virtual-Universe-NXG-master\OpenSim\Server\Base\ServerUtils.cs	44	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.ClientStack.dll' could not be found	OpenSim.Region.ClientStack.LindenUDP	D:\Virtual-Universe-NXG-master\OpenSim\Region\ClientStack\Linden\UDP\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.Framework.dll' could not be found	OpenSim.Region.ClientStack.LindenUDP	D:\Virtual-Universe-NXG-master\OpenSim\Region\ClientStack\Linden\UDP\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Server.Base.dll' could not be found	OpenSim.Server.Handlers	D:\Virtual-Universe-NXG-master\OpenSim\Server\Handlers\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.Framework.dll' could not be found	OpenSim.Services.Connectors	D:\Virtual-Universe-NXG-master\OpenSim\Services\Connectors\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Server.Base.dll' could not be found	OpenSim.Services.Connectors	D:\Virtual-Universe-NXG-master\OpenSim\Services\Connectors\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.ClientStack.LindenUDP.dll' could not be found	OpenSim.Region.CoreModules	D:\Virtual-Universe-NXG-master\OpenSim\Region\CoreModules\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.Framework.dll' could not be found	OpenSim.Region.CoreModules	D:\Virtual-Universe-NXG-master\OpenSim\Region\CoreModules\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Server.Base.dll' could not be found	OpenSim.Region.CoreModules	D:\Virtual-Universe-NXG-master\OpenSim\Region\CoreModules\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Server.Handlers.dll' could not be found	OpenSim.Region.CoreModules	D:\Virtual-Universe-NXG-master\OpenSim\Region\CoreModules\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Services.Connectors.dll' could not be found	OpenSim.Region.CoreModules	D:\Virtual-Universe-NXG-master\OpenSim\Region\CoreModules\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.ClientStack.dll' could not be found	OpenSim	D:\Virtual-Universe-NXG-master\OpenSim\Region\Application\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.CoreModules.dll' could not be found	OpenSim	D:\Virtual-Universe-NXG-master\OpenSim\Region\Application\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Region.Framework.dll' could not be found	OpenSim	D:\Virtual-Universe-NXG-master\OpenSim\Region\Application\CSC	1	Active

Code	Description	Project	File	Line	Suppression State
CS0006	Metadata file 'D:\Virtual-Universe-NXG-master\VirtualUniverse\bin\OpenSim.Server.Base.dll' could not be found	OpenSim	D:\Virtual-Universe-NXG-master\OpenSim\Region\Application\CSC	1	Active

The compiler gave  me 17 errors and  0 Warnings 

OpenSim.exe and Robust where not created.

Then I did the same with Halcyon (Compile completed) and Opensim 9.1 (Compile completed) then an older version from Virtual-Universe-NXG-master (30 days old) (Compile completed), this  gave  me the indication that it is/was  not Microsoft Visual Studio 2017, bellow a screenshot from Microsoft Visual Studio 2017 due these list. 

Regards,
Ana Green

![Error list](https://user-images.githubusercontent.com/36156774/61014678-ed158100-a388-11e9-9b21-c0cf0677666f.png)

**Comment by BruceStarfinder** [comment]
Good evening,

As I stated a few minutes ago I already reverted the commit that caused this.  Its not a true error just an annoyance caused by an expected behavior of the Mono.Addins  system.  

When attempting to compare whether the issues with a build are related to the project solution's cache or not we don't rely on whether other projects that are not ours can or cannot build.  Their build processes are often different then ours.  So therefore that really wouldn't be a legitimate test.  

For the moment this issue should now be resolved until I rewrite the versioning system to resolve the mono annoyances.

Best Regards,
Bruce Starfinder
Core Developer
Work Group member
Grid Architecture Development Work Group
Grid Architecture Development, Database Development, Maintenance
Second Galaxy Development Team
